parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -200,103 +201,141 @@ downloadSecureTLSBootstrapKubeletExecPlugin() {
+    local shims_to_download=${4}
+    local binary_version="$(echo "${shim_version}" | tr . -)" # replaces . with - == 1.2.3 -> 1-2-3
+
+    if wasmFilesExist "$containerd_wasm_filepath" "$shim_version" "$shims_to_download" "-v1"; then
+        return
+    fi
+
+    # Oras download for WASM for Network Isolated Clusters
     BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER:=}"
-    
-    for shim_version in $CONTAINERD_WASM_VERSIONS; do
-        binary_version="$(echo "${shim_version}" | tr . -)"
-
-        local version_suffix
-        local shims_to_download=()
-        local shim_prefix="containerd-shim-"
-        local registry_path
-        local base_path
-        local shim_filename
-
-        # figure out version suffix, shims to download, and paths
-        if [ "$shim_version" == "v0.15.1" ]; then
-            version_suffix="-v2"
-            shims_to_download=("spin")
-            registry_path="oss/binaries/spinkube/containerd-shim-spin"
-            base_path="spinkube"
-            shim_filename="containerd-shim-spin-v2"
-        else
-            version_suffix="-v1"
-            shims_to_download=("spin" "slight")
-            registry_path="oss/binaries/deislabs/containerd-wasm-shims"
-            base_path="containerd-wasm-shims"
-            shim_filename="containerd-wasm-shims-linux-${CPU_ARCH}.tar.gz"
-            if [ "$shim_version" == "v0.8.0" ]; then
-                shims_to_download+=("wws")
-            fi
-        fi
+    if [[ ! -z ${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER} ]]; then
+        local registry_url="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER}/oss/binaries/deislabs/containerd-wasm-shims:${shim_version}-linux-${CPU_ARCH}"
+        local wasm_shims_tgz_tmp=$containerd_wasm_filepath/containerd-wasm-shims-linux-${CPU_ARCH}.tar.gz
+
+        retrycmd_get_tarball_from_registry_with_oras 120 5 "${wasm_shims_tgz_tmp}" ${registry_url} || exit $ERR_ORAS_PULL_CONTAINERD_WASM
+        tar -zxf "$wasm_shims_tgz_tmp" -C $containerd_wasm_filepath
+        mv "$containerd_wasm_filepath/containerd-shim-*-${shim_version}-v1" "$containerd_wasm_filepath/containerd-shim-*-${binary_version}-v1"
+        rm -f "$wasm_shims_tgz_tmp"
+        return
+    fi
 
-        # check if shims are already downloaded
-        shims_missing=false
-        for shim in "${shims_to_download[@]}"; do
-            if [ ! -f "${containerd_wasm_filepath}/${shim_prefix}${shim}-${binary_version}${version_suffix}" ]; then
-                shims_missing=true
-                break
-            fi
-        done
+    for shim in "${shims_to_download[@]}"; do
+        retrycmd_if_failure 30 5 60 curl -fSLv -o "$containerd_wasm_filepath/containerd-shim-${shim}-${binary_version}-v1" "$containerd_wasm_url/containerd-shim-${shim}-v1" 2>&1 | tee $CURL_OUTPUT >/dev/null | grep -E "^(curl:.*)|([eE]rr.*)$" && (cat $CURL_OUTPUT && exit $ERR_KRUSTLET_DOWNLOAD_TIMEOUT) &
+        WASMSHIMPIDS+=($!)
+    done
+}
```

Comment:
question: shouldn't this for loop be put in the `else` block since it's downloading from `acs-mirror` path, instead of MCR registry?

Reply:
It doesn't need to be since I put a return on line 268 if it goes into the download via oras if block